### PR TITLE
Feature/multiple wishlists

### PR DESCRIPTION
**Pull request by topernicus**
image
image
image
image

### CODE REVIEW

File: src/pages-dynamic/colorway.js [hunk 1]
```diff
@@ -26,10 +26,30 @@ const Maker = (props) => {
     setState({ text: 'Copied!' });
   };
 
-  const [wishlist, setStateWishlist] = useState(undefined);
+  const [wishlistContainer, setStateWishlist] = useState({
```
Comment (zekth) [inline src/pages-dynamic/colorway.js:29]: why not setting the state to undefined? Because we need to access `container.wishlists` ?

**Review by theosenoussaoui — COMMENTED** [review]
There are multiple times in the `src/pages/wishlist.js` when you keep using the spread syntax and bracketing at the same time, and I can't figure out why, sorry. Why not returning the object itself ? 

States are almost always objects, so using the spread syntax is often unnecessary. Sorry if I missed anything !

File: src/internal/wishlist.js [hunk 2]
```diff
@@ -59,62 +67,112 @@ export function canAdd(type, wishlist) {
   }
 }
 
-export function getDefaultWishlist() {
-  return { ...defaultWishlist };
+export function getDefaultWishlistContainer() {
+  return { ...defaultWishlistContainer };
```
[the remark this answers is not in the document]
Comment (theosenoussaoui) [reply src/internal/wishlist.js]: Yeah because you have to change the imports in those files as well, obviously.

File: src/pages/wishlist.js [hunk 3]
```diff
@@ -197,7 +199,7 @@ const Wishlist = () => {
             <button
               id="addWishlist"
               onClick={addNewWishlist}
-              className={`w-2/3 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded mt-7 ${
+              className={`w-2/3 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded mt-7 disabled:bg-gray-600${
```
Comment (theosenoussaoui) [inline src/pages/wishlist.js:202]: Don't forget the space between `disabled:bg-gray-600` and `${`.
Same on line 214.

File: src/internal/wishlist.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (theosenoussaoui) [reply src/internal/wishlist.js]: I tested it locally and everything is working fine on my end (even tested in private browsing). I can add and delete sculpts and colorways without issues.

Are you sure the imports and exports are correct ? IMO that's the only way you can get an `undefined` object. You have to modify these 3 files : 

- `src/pages/wishlist.js`
- `src/pages-dynamic/sculpt.js`
- `src/pages-dynamic/colorway.js`

If it still doesn't work properly, you can always commit so I can take a look at it and find out what the problem is. :eyes:

**Review by theosenoussaoui — APPROVED** [review]

**Review by zekth — APPROVED** [review]
Let's go!!!

**Pull request merged**